### PR TITLE
[LTR] Add tom bombadil

### PR DESCRIPTION
**Pull request by alexander-novo**
Builds on top of #10130

I also added a couple of helper classes in `CountersOnPermanentsCondition` and `CountersOnPermanentsHint`, since there's one other creature with a similar ability, and it seemed like something that could come back in the future. The other creature is Backstreet Bruiser, which looks for all types of counters instead of a specific type, so it doesn't actually work with the new helper class, but maybe it's easily adjusted?

No unit tests, but I did test a couple of corner cases in games: specifically if the saga leaves the battlefield or transforms before the ability resolves, and if a counter is removed before the ability resolves.

### CODE REVIEW

**Comment by alexander-novo** [comment]
[[Tom Bombadil]] [[Backstreet Bruiser]]

**Comment by theelk801** [comment]
can you rebase this to the current master branch? I haven't had a chance to review it yet

**Comment by alexander-novo** [comment]
It looks like the pull request that added LTR that I was basing this pull request off of wasn't what ended up being merged in, so there were some merge conflicts. I pushed a merge commit to fix those.

**Comment by theelk801** [comment]
you should probably delete this branch and remake it so that it's just based on the current master

**Comment by alexander-novo** [comment]
Okay, I apologise, I've never done a non-trivial rebase before, so I decided to look up how to do it and ended up with a much cleaner branch.

File: Mage.Sets/src/mage/cards/t/TomBombadil.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        for (Card card : controller.getLibrary().getCards(game)) {
+            toReveal.add(card);
+            if (card.hasSubtype(SubType.SAGA, game)) {
+                toBattlefield = card;
+                break;
+            }
+        }
+        controller.revealCards(source, toReveal, game);
+        if (toBattlefield != null) {
+            toReveal.remove(toBattlefield);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/t/TomBombadil.java:182]: instead of removing the card here, move it to the battlefield and then use `toReveal.retainZone(Zone.Library, game)` so you don't have to worry about whether it actually went to the battlefield or some other effect prevented it

**Comment by theelk801** [comment]
no worries, I don't think I've ever done that kind of rebase either and would have just started a new branch but this definitely works

**Review by theelk801 — CHANGES_REQUESTED** [review]
just one nitpick and we're good to go

File: Mage.Sets/src/mage/cards/t/TomBombadil.java (hunk 1 above)
Comment (alexander-novo) [reply Mage.Sets/src/mage/cards/t/TomBombadil.java:182]: Okay, sounds good. I copy-pasted this wholesale from [`PrismaticBridgeEffect`](https://github.com/magefree/mage/blob/b1b0cd7d40bce897937a5a4ec04a859a63a0fa31/Mage.Sets/src/mage/cards/e/EsikaGodOfTheTree.java#L111), is this something that should change there too?

Comment (theelk801) [reply Mage.Sets/src/mage/cards/t/TomBombadil.java:182]: yeah, although it should really all be refactored to use a common class and use the `PutCards` enum which is outside the scope of this PR

**Comment by theelk801** [comment]
thanks for the contribution!
